src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -229,15 +229,19 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude) || mode == Mode.AppendVersion)
+            if (!string.IsNullOrEmpty(Href))
```

Comment:
I followed up in comments that are still visible. But FYI what you want to avoid is trouncing changes `OtherLinkTagHelper` has made already to `output.Attributes[HrefAttributeName]`. The problematic "side effect" would be that tag helper not working or not doing everything it's supposed to do.